File: 941-sort-array-by-parity/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        i = 0
        j = len(nums) - 1

        def isEven(number):
            return number % 2 == 0

        while (i <= j):
            if (isEven(nums[i])):
                i += 1
            elif (not isEven(nums[j])):
                j -= 1
            else:
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j -= 1
        
        return nums
        

File: 941-sort-array-by-parity/sort-array-by-parity.js
var sortArrayByParity = function(nums) {
    
    function isOdd(num) {
        return num % 2 != 0
    }

    i = 0
    j = nums.length - 1

    while (i < j) {
        if (!isOdd(nums[i])) i += 1
        else if (isOdd(nums[j])) j -= 1
        else {
            [nums[i], nums[j]] = [nums[j], nums[i]]
            i += 1
            j -= 1
        }
    }   

    return nums 
};